languages/python/exercises/concept/dicts/.meta/example.py

```diff
@@ -1,19 +1,22 @@
 def create_inventory(items):
     inventory = dict()
-	add_items(inventory, items)
+    add_items(inventory, items)
     return inventory
 
+
 def add_items(inventory, items):
     for item in items:
         inventory.setdefault(item, 0)
         inventory[item] += 1
     return inventory
 
+
 def delete_items(inventory, items):
     for item in items:
         inventory.setdefault(item, 0)
         inventory[item] = max(inventory[item] - 1, 0)
     return inventory
 
+
 def list_inventory(inventory):
     return [(k, v) for k, v in sorted(inventory.items()) if v]
```

Comment:
I love comprehensions, and this would be the way I would naturally write this function (_although I wouldn't use single letter variable names..._).  

**BUT** .... comprehensions are neither a prerequisite for this exercise, nor are they taught within it.  @ErikSchierboom @cmccandless -- is it ok to use a comprehension here in the `example.py` -- or should we use the loop-append method?  **e.g.**

```
def list_inventory(inventory):
    items_in_inventory = []

    for key, value in inventory.items():
        if value:
            items_in_inventory.append((key, value))
    
    return sorted(items_in_inventory)
```

Reply:
I see no problem with it:
1. This is only the example solution that verifies the tests in CI. The student won't be presented with this code
2. If a student has received an incomplete education in a language, and knows how to use some advanced features (like comprehensions), but lacks other core concepts, Exercism is still the right place for them

@ErikSchierboom , is there any requirement that the example solution be one that passes the analyzer without eliciting any criticism?

Reply:
I'm on the fence.  According to the [docs](https://exercism.github.io/v3/#/docs/concept-exercises), we should have code here that's _idiomatic_ -- but it also states that it should only be code/syntax that's been introduced previously or in prerequisites:

![image](https://user-images.githubusercontent.com/5923094/87794076-f7b7f100-c7fa-11ea-91bd-bf47f8150bf0.png)

Reply:
Should I just change the code so that it fits in the prerequisites?

Reply:
> This is only the example solution that verifies the tests in CI. The student won't be presented with this code

Well, this used to be the case in v2, but not anymore in v3. We even have some [documentation on the example file](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#example-implementation-file) nowadays. The key aspect there is that the example file should only use language features introduced by the exercise or its prerequisites (and their prerequisites, and so on). The example file will be used by mentors as a reference of the idiomatic solution of the exercise, it is not just for CI.

> I'm on the fence. According to the docs, we should have code here that's idiomatic -- but it also states that it should only be code/syntax that's been introduced previously or in prerequisites:

Yes, I've now had two people not understanding the docs, so I'll improve them!

> should we use the loop-append

My vote would be for this.